### PR TITLE
Fix rectangle-line overlap

### PR DESCRIPTION
**Pull request by nspotrepka**
Previous implementation failed for cases where both endpoints are outside of rectangle. Hard bug to notice when line segments are small.

### CODE REVIEW

**Comment by sonoro1234** [comment]
Could you please provide image of case failing. I am not able to reproduce.

**Comment by nspotrepka** [comment]
Here is one example:

one_example

**Comment by sonoro1234** [comment]
Thanks, I see now.

File: imnodes.cpp [hunk 1]
```diff
@@ -455,73 +455,51 @@ inline int sign(float val) { return int(val > 0.0f) - int(val < 0.0f); }
 
 inline bool rectangle_overlaps_line_segment(const ImRect& rect, const ImVec2& p1, const ImVec2& p2)
 {
-    if (rect.Contains(p1) && rect.Contains(p2))
```
Comment (Nelarius) [inline imnodes.cpp:458]: Ouch, I could have used a code review for this function it seems 😄

**Comment by Nelarius** [comment]
Nice work @nspotrepka , and massive props for finding this bug in the first place 👌 I'll take it.

**Pull request merged**